core/domain/blog_services.py

```diff
@@ -0,0 +1,613 @@
+    """
+    blog_post_rights = get_blog_post_rights(blog_post_id, strict=False)
+    if blog_post_rights is None:
+        raise Exception('The given blog post does not exist')
+    blog_post = get_blog_post_by_id(blog_post_id, strict=True)
+    blog_post.validate(strict=True)
+    blog_post_summary = get_blog_post_summary_by_id(blog_post_id, strict=True)
+    blog_post_summary.validate(strict=True)
+    blog_post_rights.blog_post_is_published = True
+
+    published_on = datetime.datetime.utcnow()
+    blog_post.published_on = published_on
+    blog_post_summary.published_on = published_on
+
+    save_blog_post_rights(blog_post_rights)
+    _save_blog_post_summary(blog_post_summary)
+    _save_blog_post(blog_post)
+
+
+def unpublish_blog_post(blog_post_id):
+    """Marks the given blog post as unpublished or draft.
+
+    Args:
+        blog_post_id: str. The ID of the given blog post.
+
+    Raises:
+        Exception. The given blog post does not exist.
+    """
+    blog_post_rights = get_blog_post_rights(blog_post_id, strict=False)
+    if blog_post_rights is None:
+        raise Exception('The given blog post does not exist')
+    blog_post_rights.blog_post_is_published = False
+    save_blog_post_rights(blog_post_rights)
+
+
+def delete_blog_post(blog_post_id):
+    """Deletes all the models related to a blog post.
+
+    Args:
+        blog_post_id: str. ID of the blog post which is to be
+            deleted.
+    """
+    blog_models.BlogPostModel.get(blog_post_id).delete()
+    blog_models.BlogPostSummaryModel.get(blog_post_id).delete()
+    blog_models.BlogPostRightsModel.get(blog_post_id).delete()
+
+
+def _save_blog_post_summary(blog_post_summary):
+    """Saves a BlogPostSummary domain object to the datastore.
+
+    Args:
+        blog_post_summary: BlogPostSummary. The summary object for the given
+            blog post summary.
+    """
+    model = blog_models.BlogPostSummaryModel.get(
+        blog_post_summary.id, strict=False)
+    if model:
+        model.author_id = blog_post_summary.author_id
+        model.title = blog_post_summary.title
+        model.summary = blog_post_summary.summary
+        model.tags = blog_post_summary.tags
+        model.published_on = blog_post_summary.published_on
+        model.thumbnail_filename = blog_post_summary.thumbnail_filename
+        model.url_fragment = blog_post_summary.url_fragment
+    else:
+        model = blog_models.BlogPostSummaryModel(
+            id=blog_post_summary.id,
+            author_id=blog_post_summary.author_id,
+            title=blog_post_summary.title,
+            summary=blog_post_summary.summary,
+            tags=blog_post_summary.tags,
+            published_on=blog_post_summary.published_on,
+            thumbnail_filename=blog_post_summary.thumbnail_filename,
+            url_fragment=blog_post_summary.url_fragment,
+        )
+    model.update_timestamps()
+    model.put()
+
+
+def save_blog_post_rights(blog_post_rights):
```

Comment:
Any specific reason for the other two `_save*` function to be private and this one is public?

Reply:
Yup, we will need to save the blog post rights from the blog admin services as well. Other 2 are not required.